src/meta/src/manager/iceberg_compaction.rs

```diff
@@ -97,20 +102,47 @@ impl IcebergCompactionManager {
 
     pub fn record_iceberg_commit(&self, sink_id: SinkId) {
         let mut iceberg_commits = self.iceberg_commits.write();
-        let count = iceberg_commits.entry(sink_id).or_insert(0);
-        *count += 1;
+        let commit_info = iceberg_commits.entry(sink_id).or_insert(CommitInfo {
+            count: 0,
+            first_commit_time: Instant::now(),
+        });
+        commit_info.count += 1;
     }
 
-    #[allow(dead_code)]
-    pub fn get_max_iceberg_commit_sink_id(&self) -> Option<SinkId> {
+    // function to get top n iceberg commits sink ids
+    pub fn get_top_n_iceberg_commit_sink_ids(&self, n: usize) -> Vec<SinkId> {
         let iceberg_commits = self.iceberg_commits.read();
-        iceberg_commits
+        let mut sink_ids: Vec<_> = iceberg_commits.iter().collect();
+        sink_ids.sort_by(|a, b| {
+            let a_commit_count = a.1.count;
+            let b_commit_count = b.1.count;
+
+            // Sort by commit count first
+            if a_commit_count != b_commit_count {
+                return b_commit_count.cmp(&a_commit_count);
+            }
+
+            // If commit counts are equal, sort by first commit time
+            b.1.first_commit_time
+                .duration_since(a.1.first_commit_time)
+                .cmp(&std::time::Duration::from_secs(0))
+        });
+
+        // TODO: make this configurable for each sink
+        // For now, we use a fixed interval of 1 hour
+        const MIN_SINK_COMMIT_INTERVAL: u64 = 3600; // 1 hour
+
+        sink_ids
             .iter()
-            .max_by_key(|&(_, count)| count)
-            .map(|(sink_id, _)| *sink_id)
+            .filter(|(_, commit_info)| {
+                Instant::now().duration_since(commit_info.first_commit_time)
+                    > std::time::Duration::from_secs(MIN_SINK_COMMIT_INTERVAL)
```

Comment:
After the initial 1 hour is passed since the sink first commit, are we going to aggressively trigger compaction? I thought we will still trigger compaction based on some interval instead of every time `get_top_n_iceberg_commit_sink_ids` is called..

Reply:
https://github.com/risingwavelabs/risingwave/blob/af755dff8be2784c7966deb95f5e982f2d581d7e/src/meta/src/hummock/manager/compaction/compaction_event_loop.rs#L664

You’re right, so we need to actively clean up this record after sending the msg to avoid repeated execution. I don’t think this is very intuitive. 
I also considered recording **last_compaction_ts**, but this still depends on the external call.

Reply:
PTAL @hzxa21